### ISSUE TITLE
Improve storage by copying/pasting duplicate articles

### ISSUE THREAD

**Issue by nicosomb** [wallabag/wallabag#2929]
To improve performance on wallabag websites which have thousands of articles (coucou wallabag.it), we can maybe think about a Entry sharing system. 
### What do I mean with sharing system? 
Because some articles are stored dozens of times by users on wallabag.it, we can define some rules, like "if an entry is already stored in the database and it was stored more than X days ago, we can just copy/paste this line in database for the current user". 
#### Why adding the filter "it was stored more than X days ago"?
Because recent articles can change often (an article published on lemonde.fr for example can be updated to fix typos / to add news info, etc). We can tell that if an article was stored 2 weeks ago, it will no longer be modified. 
We can define this duration in the internal settings. This filter can also be disabled.
And if the user sees that the article is not up to date, he can click on "reload content" button on article view. 
### Notification 
Maybe we can prevent the user that the article was already saved by somebody (and "due to improve perfomance, bla bla") and that he can re-fetch content with the good button. 

What do you think about this feature?

**Comment by tcitworld** [comment]
First of all, I think the performance improvements should be tested to be sure it's really a good improvement.

In that case, I think we should have a attribute on the entry to see if it's original or not. Currently we only allow the title to be edited but I guess at some point the whole entry will be editable.

**Comment by nicosomb** [comment]
> First of all, I think the performance improvements should be tested to be sure it's really a good improvement.

Agree with you. 
But if we don't use graby to fetch article's content, we can easily improve performance here.

**Comment by j0k3r** [comment]
I think it won't improve _performance_ but it'll be _faster_ for the user when saving an existing article.

Maybe we can just try to find the latest article saved for that url without notifing user (with the filter of course).
I personally don't care if the _system_ fetch the content from an other user or from the real url. I just want my article to be readable.

Of course, a filter should also be used in combination with paywall content. We shouldn't retrieve a content saved by an authenticated user on LeMonde for a non-authenticated user.

Finally what happen if a user re-fetch a content: should we update all associated entry with that same url with the re-fetched content?

**Comment by tcitworld** [comment]
> Finally what happen if a user re-fetch a content: should we update all associated entry with that same url with the re-fetched content?

The best solution would be to notify the others users that some entries changed

**Comment by nicosomb** [comment]
> The best solution would be to notify the others users that some entries changed

No. If I want to save a specific version of a web page (eg: wikipedia article), I don't want that other users can update **my** version of the article.
